src/validation.cpp

```diff
@@ -1483,79 +1473,197 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
+        // In addition to resetting m_viewmempool, we also need to manually delete these coins from
+        // m_view because it caches copies of the coins it fetched from m_viewmempool previously.
+        m_view.Uncache(outpoint);
+    }
+    // This deletes the temporary and mempool coins.
+    m_viewmempool.Reset();
+
+    // Now that we have determined which transactions to skip, do full validation for each
+    // transaction with its ancestor subpackage. If this is a single transaction, call
+    // AcceptSingleTransaction() with a single transaction ATMPArgs to avoid unnecessary package
+    // restrictions like disallowment of RBF. Skip when we know the feerate will not be sufficient.
+    // If we encounter any errors at all, we quit immediately and do not validate anything else.
+    // This means we may potentially skip validation for valid transactions, but we can expect that
+    // some other peer will announce those transactions to us.
+    PackageValidationState package_state_final;
+    if (!quit_early) {
+        // Use the fee information gathered in the PreChecks() loop above to refine linearization.
+        // We want to try to submit the higher-feerate subpackages and transactions first.
+        // If linearization fails, we will only consider submitting transactions by themselves.
+        const bool rely_on_fees{linearized_package.LinearizeWithFees()};
+
+        for (const auto& tx : linearized_package.FilteredTxns()) {
+            const auto individual_fee_vsize = linearized_package.GetFeeAndVsize(tx);
+            TxValidationState placeholder_state;
+            if (rely_on_fees && Assume(individual_fee_vsize.has_value()) &&
+                !CheckFeeRate(individual_fee_vsize->second, individual_fee_vsize->first, placeholder_state)) {
+                // No need to validate if we know this transaction wouldn't meet feerate requirements.
+                // If we don't exit early here, we'll hit a low fee failure and quit early.
+                // Even if package feerate is sufficient, we don't allow "parent pay for child."
+                continue;
+            }
+            const auto subpackage_fee_vsize = linearized_package.FilteredAncestorFeeAndVsize(tx);
+            const auto subpackage = linearized_package.FilteredAncestorSet(tx);
+            if (!subpackage) continue;
+
+            // If rely_on_fees is false, we can't be sure that we're getting subpackages that are
+            // CPFPs. To avoid creating "parent pays for child" behavior, only consider
+            // 1-transaction subpackages (thus disabling package CPFP).
+            if (!rely_on_fees && subpackage->size() > 1) continue;
+
+            // No need to validate if we know this subpackage won't meet feerate
+            // requirements. If it's a CPFP'd transaction, presumably there is a subsequent
+            // subpackage that will bump it.
+            if (rely_on_fees && subpackage_fee_vsize.has_value() &&
```

Comment:
rely_on_fees implies this?
```suggestion
            if (rely_on_fees && Assume(subpackage_fee_vsize.has_value()) &&
```

Reply:
Yep